truex-ios-mobile-reference-app/VideoPlayerViewController.m

```diff
@@ -105,87 +106,110 @@ - (void)adBreakStarted {
+    NSLog(@"Ad Manager: Ad Break Ended");
     self.requiresLinearPlayback = NO;
 }
 
 // MARK: - TRUEX DELEGATE METHODS
+// [5] - Other delegate method
 - (void)onAdStarted:(NSString*)campaignName {
+    // true[X] - User has started their ad engagement
     NSLog(@"truex: onAdStarted: %@", campaignName);
 }
 
+// [4] - Respond to renderer terminating events
 - (void)onAdCompleted:(NSInteger)timeSpent {
+    // true[X] - User has finished the true[X] engagement, resume the video stream
     NSLog(@"truex: onAdCompleted: %ld", (long)timeSpent);
     [self resetActiveAdRenderer];
     [self.player play];
 }
 
+// [4]
 - (void)onAdError:(NSString*)errorMessage {
+    // true[X] - TruexAdRenderer encountered an error presenting the ad, resume with standard ads
     NSLog(@"truex: onAdError: %@", errorMessage);
     [self resetActiveAdRenderer];
     [self.player play];
 }
 
+// [4]
 - (void)onNoAdsAvailable {
+    // true[X] - TruexAdRenderer has no ads ready to present, resume with standard ads
     NSLog(@"truex: onNoAdsAvailable");
     [self resetActiveAdRenderer];
     [self.player play];
 }
 
+// [3] - Respond to onAdFreePod
 - (void)onAdFreePod {
+    // true[X] - User has met engagement requirements, skips past remaining pod ads
     NSLog(@"truex: onAdFreePod");
-    if (_resumeTime != -1) {
-        // Custom resume logic for snap back
+    if (_resumeTime == -1) {
+        // true[X] - Skipping the whole ad break here as user earn credit from true[X]
+        [self seekOverCurrentAdBreak];
+    } else {
+        // Custom snap back logic, skipping ad break and send user back to their original position
         [self.player seekToTime:CMTimeMake(_resumeTime, 1)];
         _resumeTime = -1;
-    } else {
-        [self seekOverCurrentAdBreak];
     }
     [self helperEndAdBreak];
 }
 
+// [5] - Other delegate method
 - (void)onPopupWebsite:(NSString *)url {
+    // true[X] - User wants to open an external link in the true[X] ad
+
     NSLog(@"truex: onPopupWebsite: %@", url);
     // Open the URL in Safari
     // [[UIApplication sharedApplication] openURL:[NSURL URLWithString: url] options:@{} completionHandler:nil];
     
-    // Or open with your existing in app webview
+    // Or, open with the existing in-app webview
     UIStoryboard* storyBoard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
     WebViewViewController* newViewController = [storyBoard instantiateViewControllerWithIdentifier:@"webviewVC"];
     newViewController.url = [NSURL URLWithString:url];
     newViewController.modalPresentationStyle = UIModalPresentationPopover;
     __weak typeof(self) weakSelf = self;
     newViewController.onDismiss = ^(void) {
+        // true[X] - You will need to pause and remume the true[X] Ad Renderer
         [weakSelf.activeAdRenderer resume];
     };
     [self.activeAdRenderer pause];
     [self presentViewController:newViewController animated:YES completion:nil];
 }
 
-// @optional
+// @optional true[X] delegate methods
```

Comment:
Should we use `// MARK:` for sections like this: https://www.avanderlee.com/xcode/xcode-mark-line-comment/